src/components/accordion/Accordion.jsx

```diff
@@ -80,14 +88,53 @@ const Accordion = ({
   className = '',
   spacing = 's',
   reduceMotion = false,
+  index,
+  onChange,
 }: AccordionPropsType) => {
   const wrapperRef = useRef<HTMLDivElement | null>(null);
+  const isControlled = useRef<boolean>(index !== undefined && !!onChange);
   const [state, dispatch] = useReducer(reducer, {
     opened: {},
     focusedElementId: null,
   });
   const hasReduceMotion = useReducedMotion() || reduceMotion;
 
+  useEffect(() => {
+    const isCallbackMissing = index !== undefined && !onChange;
+    const isComponentChangedToUncontrolled = isControlled.current && !index;
```

Comment:
I think we need to be more explicit here - `typeof index !== "undefined" - to avoid false positives (`index = 0`).